src/components/TabBar.tsx

```diff
@@ -81,7 +90,21 @@ const TabContainer = styled.div`
   display: flex;
 `
 
-const Tab = styled.div<{ isActive: boolean }>`
+const TabSelector = styled(motion.div)`
+  position: absolute;
+  bottom: 0;
+  height: 2px;
+  border-radius: var(--radius);
+  flex: 1;
+  background-color: ${({ theme }) => theme.global.accent};
+  z-index: -1;
+`
+
+const Tab = styled(({ isActive, onClick, onKeyPress, children, className }) => (
+  <div className={className} onClick={onClick} onKeyPress={onKeyPress} role="tab" tabIndex={0} aria-selected={isActive}>
+    {children}
+  </div>
+))`
```

Comment:
That's an interesting approach..!

The alternative would take more lines to implement, so I kinda like this one. @mvaivre WDYT?

Reply:
I personally prefer the usual separation of concerns :

```
// Components & logic first
<Components />
...
// Styles at the bottom
const styled(...)
```

Even though this solution is shorter, it may make the code less readable in the end IMHO. 
I've seen that you used a similar technique in `HiddenLabel` and `Truncate`, but without naming the components. So if I don't look at the file name, it's not easy to understand what the component do. And I find it less easy to parse the props they take, as the code is more cluttered, with the styling glued to the logic.

Reply:
The only way to know what a component does is to read its code - take `AddressBadge` for example. It's not possible to know what it does just based on its name... Rather it identifies what behavior its associated with, so regardless, you will have to read the documentation (or code if lack thereof)...

Yep the code is more cluttered because our format rules enforce those props to be all on one line. :shrug: 

I don't mind either way. The complexity of the code is so minimal that written either way could be interpreted as easy to understand. Just let me know what you both would like. :+1:

Reply:
>  Take `AddressBadge` for example. It's not possible to know what it does just based on its name

Well, I don't agree with that: if that constant is called `AddressBadge`, I know that it's supposed to be use as an address badge.

Now, when you read the following, how do you know what it's supposed to be used for ? 

```ts
export default styled(({ text, ...props }) => <div {...props}>{text}</div>)`
  position: absolute;
  left: -10000px;
  top: auto;
  width: 1px;
  height: 1px;
  overflow: hidden;
`
```

----------

> Yep the code is more cluttered because our format rules enforce those props to be all on one line.

It's not the one-liner that's annoying me, it's the lack of blank lines. Maybe it's just me but when I read & write some code, I use blank line to give the file some structure. Ie. I assume that two pieces of code with different goals are gonna be split by one or multiple blank lines. Here, logic and style are sticked together, and that's what I don't really like. 

But let's avoid wasting too much time on those things. @nop33, what's you final opinion on this? I honestly don't really care if some components are written that way, but in the long run, I'd prefer if we would stick to our previous ways which were working well for us. In the end, all I want is for us to write consistent, readable code :)

Reply:
I get the arguments from both of you and I don't have a super strong preference, but I tend to go with @mvaivre's view.